systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMaker2IsolatedST.java

```diff
@@ -1633,6 +1635,39 @@ void testKMM2RollAfterSecretsCertsUpdateTLS(ExtensionContext extensionContext) {
         LOGGER.info("Messages successfully mirrored");
     }
 
+    @ParallelNamespaceTest
+    void testMirrorMaker2RackAwareness(ExtensionContext extensionContext) {
+        final String namespaceName = StUtils.getNamespaceBasedOnRbac(INFRA_NAMESPACE, extensionContext);
+        String clusterName = mapWithClusterNames.get(extensionContext.getDisplayName());
+        String kafkaClusterSourceName = clusterName + "-source";
+        String kafkaClusterTargetName = clusterName + "-target";
+        String topologyKey = "kubernetes.io/hostname";
+
+        resourceManager.createResource(extensionContext,
+                KafkaTemplates.kafkaEphemeral(kafkaClusterSourceName, 1, 1).build());
+        resourceManager.createResource(extensionContext,
+                KafkaTemplates.kafkaEphemeral(kafkaClusterTargetName, 1, 1).build());
+
+        resourceManager.createResource(extensionContext,
+                KafkaMirrorMaker2Templates.kafkaMirrorMaker2(clusterName, kafkaClusterTargetName, kafkaClusterSourceName, 1, false)
+                    .editSpec()
+                        .withNewRack(topologyKey)
+                    .endSpec()
+                    .build());
+
+        LOGGER.info("MirrorMaker2 cluster deployed successfully");
+        String podName = PodUtils.getPodNameByPrefix(namespaceName, KafkaMirrorMaker2Resources.deploymentName(clusterName));
+        String podJson = TestUtils.toJsonString(kubeClient(namespaceName).getPod(namespaceName, podName));
+        assertThat(podJson, hasJsonPath(StUtils.initGlobalVariableJsonPathBuilder(0, "RACK_TOPOLOGY_KEY"), hasItem(topologyKey)));
+        assertThat(podJson, hasJsonPath(StUtils.nodeSelectorKeyJsonPathBuilder(topologyKey), hasItem(topologyKey)));
+
+        String podLogs = kubeClient(namespaceName).logs(podName);
+        String podNodeName = PodUtils.getPodByName(namespaceName, podName).getSpec().getNodeName();
+        String hostname = podNodeName.contains(".") ? podNodeName.substring(0, podNodeName.indexOf(".")) : podNodeName;
+        String clientRackConfig = "consumer.client.rack=" + hostname;
+        assertTrue(podLogs.contains(clientRackConfig), "Pod logs contain " + clientRackConfig);
```

Comment:
Why are you using the JSON output for assertion etc? It can be done using the fabric8 api as in other rack awareness tests.

Reply:
OK, I'll look at that. Thanks.